### PR TITLE
fix: class extends itself #2201

### PR DESCRIPTION
**Pull request by technohippy**
⯈fix: #2201 

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/diagnosticMessages.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
   "Property '{0}' is always assigned before being used.": 233,
   "Expression does not compile to a value at runtime.": 234,
   "Only variables, functions and enums become WebAssembly module exports.": 235,
+  "Circular inheritance": 236,
```
Comment (romdotdog) [inline src/diagnosticMessages.json:49]: It's better to use TypeScript's errors if possible; i.e. add and use TS2506 for this
`"'{0}' is referenced directly or indirectly in its own base expression": 2506`

**Review by romdotdog — COMMENTED** [review]
Thank you for the PR.

File: src/diagnosticMessages.json (hunk 1 above)
Comment (technohippy) [reply src/diagnosticMessages.json]: Thanks, done.

**Comment by dcodeIO** [comment]
I think `markVirtuals` may not be the ideal place for the check, as it is doing something rather specific at the end of compilation. A better place would be where `.base` is set in a way that the error condition appears, producing a diagnostic then, so `markVirtuals` wouldn't need to check?

**Comment by dcodeIO** [comment]
Thank you!

**Pull request merged**